src/components/StoryPointsContainer.tsx

```diff
@@ -0,0 +1,107 @@
+'use client'
+
+import { socketRoomEmitter } from '@/services/socket'
+import { useSocketListener } from '@/services/socket'
+import { useSocketListenerWithCb } from '@/services/socket'
+import { useState, useEffect } from 'react'
+import StoryPointButton from '@/components/StoryPointButton'
+import type { RoomInfoData } from '@/components/RoomMainUi'
+import type { ListenerRes } from '@/services/socket'
+
+export default function StoryPointsContainer({
+	roomId,
+	userName,
+}: RoomInfoData) {
+	const [selectedStoryPoint, setSelectedStoryPoint] = useState<string | null>(
+		null,
+	)
+	const [isPointBtnDisabled, setIsPointBtnDisabled] = useState(false)
+
+	// const resetStoryPointBtns = useSocketListener('show-disable-reset-points')
+	const resetStoryPointBtns = useSocketListenerWithCb(
+		'show-disable-reset-points',
+		{
+			callback: (showDisableReset: ListenerRes) => {
```

Comment:
We can have many types of callbacks in this listner. Imagine, `onError`, `onLoading`, etc. This one should be called `onEmit`

Reply:
I've modified it to just be "callback" so that it can take anything. If I have to change the name, then I have to account for all of those in the socket.tsx file.

Reply:
> Then I just have lots of "onThis, onThat" which seems like the opposite of what I should do. I'd rather call it what it is: callback.

But having explicit onThis, onThat, etc, makes the code simpler to manage and understand. Another reason to use an object for the Config/Options is that it allow us to extend with more stuff more easily.

In other words, tell me, which code is easier to read:

```js
// Example 1
const isGameOver = true;

// Example 1
useEvent('join-game', (value) => {
   console.log(value)
}, isGameOver)

// Example 2
useEvent('join-game', {
   onEmit: (value) => {
     console.log(value)
   }
  disable: isGameOver
}
```

The 1st example, we know there's a callback, but we don't know when it is executed. Then we have "isGameOver" (true) as last parameter but we don't understand what it does.

The 2nd example, we know the callback is executed when the emit happens, and we understand the isGameOver is used to disable the event (a hypothetical possible feature).

Using an Object to pass optional parameters is a very common pattern. Please read the 2 articles about it: [The options object pattern (short version)](https://medium.com/@bchadwickfrance/the-options-object-pattern-with-typescript-a14f33306ec8) and [The options object pattern (more detailed)https://michaelnthiessen.com/options-object-pattern).

Reply:
The problem I'm thinking about is in the socket.tsx file. I'm thinking there will have to be a line for `onEmit`, a line for `onChange`, etc., and duplicating the code. The generic name seemed more concise. 

I'll read the articles. We will have to go over the above since I'm not sure at all how that would look in the socket.tsx file.

Reply:
I've changed the 3 places that have a listener and a callback to use the "options object pattern". I sort of get it, but I still need to read the 2 articles. I will read them in the morning. 

The type for Config is:

```
type Config = {
     onChange: (data: ListenerRes) => void
}
```
Thank you Copilot!! LOL

DONE

Reply:
> a line for onEmit, a line for onChange, etc., and duplicating the code. 

You won't be duplicating the code because each case does different things/code, as you need it. Don't worry about it for now. Trust the pattern and you'll see benefits in the future ;)